libs/wingc/src/type_check_assert.rs

```diff
@@ -0,0 +1,13 @@
+use crate::{ast::Expr, visit::Visit};
+
+pub struct TypeCheckAssert;
+
+impl Visit<'_> for TypeCheckAssert {
+	fn visit_expr(&mut self, expr: &Expr) {
+		assert!(
+			expr.evaluated_type.borrow().is_some(),
+			"Expr was not type checked: {:?}",
+			expr
+		);
+	}
```

Comment:
Do you need to also call the global visit_expr to make sure this recurses properly?

Reply:
Took a little bit of work to add this because it surfaced some more bugs where types weren't being evaluated. 😅